### PR TITLE
provider/google: Add CORS support for google_storage_bucket.

### PR DESCRIPTION
**Pull request by rileykarson**
Add support for CORS configuration fields to google_storage_bucket.

```
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/05/19 16:45:10 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccStorage -timeout 120m
=== RUN   TestAccStorage_basic
--- PASS: TestAccStorage_basic (2.90s)
=== RUN   TestAccStorageCustomAttributes
--- PASS: TestAccStorageCustomAttributes (6.12s)
=== RUN   TestAccStorageStorageClass
--- PASS: TestAccStorageStorageClass (7.09s)
=== RUN   TestAccStorageBucketUpdate
--- PASS: TestAccStorageBucketUpdate (8.40s)
=== RUN   TestAccStorageBucketImport
--- PASS: TestAccStorageBucketImport (2.82s)
=== RUN   TestAccStorageForceDestroy
Created object bucketDestroyTestFile at location https://www.googleapis.com/storage/v1/b/tf-test-acl-bucket-4919795548258768973/o/bucketDestroyTestFile

--- PASS: TestAccStorageForceDestroy (14.62s)
=== RUN   TestAccStorage_cors
--- PASS: TestAccStorage_cors (2.78s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/google	44.852s
```

Fix for #12519.
@danawillow

### CODE REVIEW

File: website/source/docs/providers/google/r/storage_bucket.html.markdown [hunk 1]
```diff
@@ -50,15 +50,27 @@ to `google_storage_bucket_acl.predefined_acl`.
...
 * `not_found_page` - (Optional) The custom object to return when a requested
     resource is not found.
+    
+The `cors` block supports:
+
+* `origin` - (Optional) The list of [Origins](https://tools.ietf.org/html/rfc6454) eligible to receive CORS response headers. Note: "*" is permitted in the list of origins, and means "any Origin".
+    
+* `method` - (Optional) The list of HTTP methods on which to include CORS response headers, (GET, OPTIONS, POST, etc) Note: "*" is permitted in the list of methods, and means "any method".
+    
+* `response_header` - (Optional) 	The list of HTTP headers other than the [simple response headers](https://www.w3.org/TR/cors/#simple-response-header) to give permission for the user-agent to share across domains.
```
Comment (danawillow) [inline website/source/docs/providers/google/r/storage_bucket.html.markdown]: nit: too many spaces?

File: builtin/providers/google/resource_storage_bucket_test.go [hunk 2]
```diff
@@ -302,3 +321,17 @@ resource "google_storage_bucket" "bucket" {
 }
 `, bucketName, storageClass, locationBlock)
 }
+
+func testGoogleStorageBucketsCors(bucketName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+	name = "%s"
+	  cors {
```
Comment (rileykarson) [inline builtin/providers/google/resource_storage_bucket_test.go]: Done.

File: website/source/docs/providers/google/r/storage_bucket.html.markdown (hunk 1 above)
Comment (rileykarson) [inline website/source/docs/providers/google/r/storage_bucket.html.markdown]: Done.

File: builtin/providers/google/resource_storage_bucket.go [hunk 3]
```diff
@@ -89,6 +89,40 @@ func resourceStorageBucket() *schema.Resource {
 					},
 				},
 			},
+
+			"cors": &schema.Schema{
+				Type:     schema.TypeSet,
```
Comment (rileykarson) [inline builtin/providers/google/resource_storage_bucket.go]: No reason really - List semantics weren't important to it, and the first similar block I looked at was a Set. Changed to List for consistency with the rest of the file; it also ends up simplifying a type assertion that way.

File: builtin/providers/google/resource_storage_bucket.go [hunk 4]
```diff
@@ -295,3 +338,41 @@ func resourceStorageBucketStateImporter(d *schema.ResourceData, meta interface{}
 	d.Set("name", d.Id())
 	return []*schema.ResourceData{d}, nil
 }
+
+func expandCors(configured []interface{}) []*storage.BucketCors {
+	corsRules := make([]*storage.BucketCors, 0, len(configured))
+	for _, raw := range configured {
+		data := raw.(map[string]interface{})
+		corsRule := storage.BucketCors{}
```
Comment (danawillow) [inline builtin/providers/google/resource_storage_bucket.go]: style nit: totally just my opinion but I think filling out structs/maps inline looks a bit better:
```
		corsRule := storage.BucketCors{
			Origin:         convertSchemaArrayToStringArray(data["origin"].([]interface{})),
			Method:         convertSchemaArrayToStringArray(data["method"].([]interface{})),
			ResponseHeader: convertSchemaArrayToStringArray(data["response_header"].([]interface{})),
			MaxAgeSeconds:  int64(data["max_age_seconds"].(int)),
		}
```
and below:
```
		data := map[string]interface{}{
			"origin":          corsRule.Origin,
			"method":          corsRule.Method,
			"response_header": corsRule.ResponseHeader,
			"max_age_seconds": corsRule.MaxAgeSeconds,
		}
```

Comment (rileykarson) [inline builtin/providers/google/resource_storage_bucket.go]: Done.

**Pull request merged**